In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statsmodels.formula.api as smf
import statsmodels.stats.api as sms

In [13]:
gdp_le = pd.read_csv('../data/gdp_le.csv')
gdp_le.head(2)

Country  Year  GDP_Per_Capita Continent  Life_Expectancy
0  Afghanistan  2021     1517.016266      Asia           61.982
1  Afghanistan  2020     1968.341002      Asia           62.575

# Problem Solving:

## 1. Start by fitting a linear regression model with target being life expectancy and predictor variable year.

In [25]:
lr_le_yr = smf.ols("Life_Expectancy ~ Year", data=gdp_le).fit()

In [27]:
lr_le_yr.params

Intercept   -468.138140
Year           0.267722
dtype: float64

Substituting these estimates into the model, we get

$$Mean(Life Expectancy) = -468.138140 + 0.267722(Year)$$

### a. What coefficients do you get? Interpret the meaning of these coefficents.

Intercept   -468.138140 and
Year           0.267722

The intercept, -468.14, is where the Life_Expectancy variable intercepts the Y axis when the year is 0. BUT the year is never 0 in this 